### ISSUE TITLE
Deskshare applet - build failure

### ISSUE THREAD

**Issue by falinsky** [bigbluebutton/bigbluebutton#3164]
When trying to build deskshare applet from branch v1.0.x-release I get next error:

`~/dev/bigbluebutton/deskshare/applet$ gradle jar --info`

> Starting Build
> Settings evaluated using settings file '/home/ubuntu/dev/bigbluebutton/deskshare/settings.gradle'.
> Projects loaded. Root project using build file '/home/ubuntu/dev/bigbluebutton/deskshare/build.gradle'.
> Included projects: [root project 'deskshare', project ':app', project ':applet', project ':common']
> Evaluating root project 'deskshare' using build file '/home/ubuntu/dev/bigbluebutton/deskshare/build.gradle'.
> Build file '/home/ubuntu/dev/bigbluebutton/deskshare/build.gradle': line 17
> The ArtifactRepositoryContainer.add(DependencyResolver, Closure) method has been deprecated and is scheduled to be removed in Gradle 2.0.
> Build file '/home/ubuntu/dev/bigbluebutton/deskshare/build.gradle': line 60
> The RepositoryHandler.mavenRepo() method has been deprecated and is scheduled to be removed in Gradle 2.0. Please use the maven() method instead.
> The scalaTools configuration has been deprecated and is scheduled to be removed in Gradle 2.0. Typically, usages of 'scalaTools' can simply be removed, and the Scala tools libraries to be used will be inferred from the Scala library found on the regular (compile) class path. In some cases, it may be necessary to additionally configure the 'scalaClasspath' property of ScalaCompile and ScalaDoc tasks.
> Evaluating project ':app' using build file '/home/ubuntu/dev/bigbluebutton/deskshare/app/build.gradle'.
> Evaluating project ':applet' using build file '/home/ubuntu/dev/bigbluebutton/deskshare/applet/build.gradle'.
> Evaluating project ':common' using build file '/home/ubuntu/dev/bigbluebutton/deskshare/common/build.gradle'.
> All projects evaluated.
> Selected primary task 'jar'
> Tasks to be executed: [task ':common:compileJava', task ':common:processResources', task ':common:classes', task ':common:jar', task ':applet:compileJava', task ':applet:processResources', task ':applet:classes', task ':applet:jar']
> :common:compileJava (Thread[main,5,main]) started.
> :common:compileJava
> Skipping task ':common:compileJava' as it is up-to-date (took 0.244 secs).
> :common:compileJava UP-TO-DATE
> :common:compileJava (Thread[main,5,main]) completed. Took 0.306 secs.
> :common:processResources (Thread[main,5,main]) started.
> :common:processResources
> Skipping task ':common:processResources' as it has no source files.
> :common:processResources UP-TO-DATE
> :common:processResources (Thread[main,5,main]) completed. Took 0.004 secs.
> :common:classes (Thread[main,5,main]) started.
> :common:classes
> Skipping task ':common:classes' as it has no actions.
> :common:classes UP-TO-DATE
> :common:classes (Thread[main,5,main]) completed. Took 0.002 secs.
> :common:jar (Thread[main,5,main]) started.
> :common:jar
> Skipping task ':common:jar' as it is up-to-date (took 0.051 secs).
> :common:jar UP-TO-DATE
> :common:jar (Thread[main,5,main]) completed. Took 0.07 secs.
> :applet:compileJava (Thread[main,5,main]) started.
> :applet:compileJava
> Cached resource is up-to-date (lastModified: Sat Jul 17 13:36:11 UTC 2010). [HTTP: http://repo1.maven.org/maven2/org/testng/testng/5.8/testng-5.8.pom]
> Cached resource is up-to-date (lastModified: Sat Jul 17 13:36:11 UTC 2010). [HTTP: http://oss.sonatype.org/content/repositories/releases/org/testng/testng/5.8/testng-5.8.pom]
> Cached resource is up-to-date (lastModified: Thu Aug 14 06:48:05 UTC 2008). [HTTP: http://repo1.maven.org/maven2/net/jcip/jcip-annotations/1.0/jcip-annotations-1.0.pom]
> Resource missing. [HTTP GET: http://repo1.maven.org/maven2/com/sun/java/plugin/1.6.0_16/plugin-1.6.0_16.pom]
> Resource missing. [HTTP HEAD: http://repo1.maven.org/maven2/com/sun/java/plugin/1.6.0_16/plugin-1.6.0_16.jar]
> Resource missing. [HTTP GET: http://oss.sonatype.org/content/repositories/snapshots/com/sun/java/plugin/1.6.0_16/plugin-1.6.0_16.pom]
> Resource missing. [HTTP HEAD: http://oss.sonatype.org/content/repositories/snapshots/com/sun/java/plugin/1.6.0_16/plugin-1.6.0_16.jar]
> Resource missing. [HTTP GET: http://oss.sonatype.org/content/repositories/releases/com/sun/java/plugin/1.6.0_16/plugin-1.6.0_16.pom]
> Resource missing. [HTTP HEAD: http://oss.sonatype.org/content/repositories/releases/com/sun/java/plugin/1.6.0_16/plugin-1.6.0_16.jar]
> :applet:compileJava (Thread[main,5,main]) completed. Took 3.632 secs.
> FAILURE: Build failed with an exception.
> What went wrong:
> Could not resolve all dependencies for configuration ':applet:compile'.
> Could not find com/sun/java:plugin:1.6.0_16.
> Required by:
> org.bigbluebutton:applet:unsigned-0.9.0

**Comment by ritzalam** [comment]
What version of Java do you have installed?

Can you change in build.gradle the following?

sourceCompatibility=1.7
targetCompatibility=1.7

**Comment by falinsky** [comment]
@ritzalam Thank you. I've changed, but the same result.

**Comment by kiransuthar6** [comment]
I am able to build applet jar with some weird steps. I solved it using java 6 and plugin.jar of oracle java7. Performed following steps.
But I think there must be some better way to solve this so let me if anyone can find. Please follow following steps.

Note : Java installation steps given bellow is for debian machine, For Ubantu it may be different.

Install java6 applying following commands(For Debian)

    echo "deb http://ppa.launchpad.net/webupd8team/java/ubuntu precise main" | sudo tee -a /etc/apt/sources.list
    echo "deb-src http://ppa.launchpad.net/webupd8team/java/ubuntu precise main" | sudo tee -a /etc/apt/sources.list
    sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys EEA14886
    sudo apt-get update
    sudo apt-get install oracle-java6-installer

There is a open(when this applet was compiled) issue of deskshare build failure https://github.com/bigbluebutton/bigbluebutton/issues/3164

Once java 6 is installed you have to set java 6 as default so gradle can use it.
    		export JAVA_HOME=/usr/lib/jvm/java-6-oracle
Now check that gradle is using java 6
            gradle -version

Now install oracle-java7 using following command
sudo apt-get install oracle-java7-installer

Copy plugin.jar in orcale java 6 jre directory, if ext folder does not exist in java 6, create it
sudo cp /usr/lib/jvm/java-7-oracle/jre/lib/plugin.jar /usr/lib/jvm/java-6-oracle/jre/lib/ext/

change the permission of plugin.jar in /usr/lib/jvm/java-6-oracle/jre/lib/ext/

 sudo chown bbb:bbb /usr/lib/jvm/java-6-oracle/jre/lib/ext/plugin.jar


Just use java6 for building applet jar and then again change to java7 otherwise your other services may not work.
 		sudo update-alternatives --config java

**Comment by helpyhelperton** [comment]
When I get to 

gradle jar

I receive the following build error

Could not resolve all dependencies for configuration ':applet:compile'.
> Could not find com/sun/java:plugin:1.6.0_16.
  Required by:
      org.bigbluebutton:applet:unsigned-0.9.0


My default Java installation is OpenJDK 7 but I have installed OpenJDK 6 and I'm executing

gradle jar -Dorg.gradle.java.home=/usr/lib/jvm/java-6-openjdk-amd64

The installed version was build 45. I tried changing the build.gradle reference to com/sun/:plugin:1.6.0_45 but that didn't help.
